egs/librispeech/asr/simple_v1/train_fast.py

```diff
@@ -0,0 +1,286 @@
+
+    return total_objf, total_frames
+
+
+def train_one_epoch(dataloader, valid_dataloader, model, device, graph,
+                    symbols, optimizer, current_epoch, num_epochs):
+    total_objf = 0.
+    total_frames = 0.
+
+    model.train()
+    for batch_idx, batch in enumerate(dataloader):
+        curr_batch_objf, curr_batch_frames = get_objf(batch, model, device,
+                                                      graph, symbols, True,
+                                                      optimizer)
+
+        total_objf += curr_batch_objf
+        total_frames += curr_batch_frames
+
+        if batch_idx % 100 == 0:
+            logging.info(
+                'processing batch {}, current epoch is {}/{} '
+                'global average objf: {:.6f} over {} '
+                'frames, current batch average objf: {:.6f} over {} frames'.
+                format(
+                    batch_idx,
+                    current_epoch,
+                    num_epochs,
+                    total_objf / total_frames,
+                    total_frames,
+                    curr_batch_objf / curr_batch_frames,
+                    curr_batch_frames,
+                ))
+
+        if valid_dataloader and batch_idx % 1000 == 0:
+            total_valid_objf, total_valid_frames = get_validation_objf(
+                dataloader=valid_dataloader,
+                model=model,
+                device=device,
+                graph=graph,
+                symbols=symbols)
+            model.train()
+            logging.info(
+                'Validation average objf: {:.6f} over {} frames'.format(
+                    total_valid_objf / total_valid_frames, total_valid_frames))
+    return total_objf
+
+
+def main():
+    # load L, G, symbol_table
+    lang_dir = 'data/lang_nosp'
+    with open(lang_dir + '/L.fst.txt') as f:
+        L = k2.Fsa.from_openfst(f.read(), acceptor=False)
+
+    with open(lang_dir + '/G.fsa.txt') as f:
+        G = k2.Fsa.from_openfst(f.read(), acceptor=True)
+
+    with open(lang_dir + '/words.txt') as f:
+        symbol_table = k2.SymbolTable.from_str(f.read())
+
+    print("start")
+    L = k2.arc_sort(L.invert_())
+    G = k2.arc_sort(G)
+    print(k2.is_arc_sorted(k2.get_properties(L)))
+    print(k2.is_arc_sorted(k2.get_properties(G)))
+    graph = k2.intersect(L, G)
+    graph = k2.arc_sort(graph)
+    print(k2.is_arc_sorted(k2.get_properties(graph)))
+
+    # load dataset
+    feature_dir = 'exp/data1'
+    print("About to get train cuts")
+    #cuts_train = CutSet.from_json(feature_dir +
+    #                              '/cuts_train-clean-100.json.gz')
+    cuts_train = CutSet.from_json(feature_dir + '/cuts_dev-clean.json.gz')
+    print("About to get dev cuts")
+    cuts_dev = CutSet.from_json(feature_dir + '/cuts_dev-clean.json.gz')
+
+    print("About to create train dataset")
+    train = K2SpeechRecognitionIterableDataset(cuts_train,
+                                               max_frames=1000,
```

Comment:
What's the benefit of using a low value like 1000 here? It'd result in batches that contain at most 10 seconds of audio.

Reply:
just for debugging/printing.  will change this once it fully runs.

Reply:
Have deleted